### PR TITLE
[MNG-7509] - Huge memory cost when parent pom widely used in a big project for dependencyManagement for 3.8.x

### PR DESCRIPTION
**Pull request by Yougoss**
Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
@Yougoss looks like you don't  `mvn clen verify` on your local host ... 😄

**Comment by Yougoss** [comment]
> @Yougoss looks like you don't `mvn clean verify` on your local host ... 😄

Hi @slawekjaranowski, I have run `mvn clean verify` but with the params '-Drat.numUnapprovedLicenses=100'

I think there are a lot of code-gen class without lisence comment. I can't run `mvn clean verify` successfully even in the current branch without any code change.

Please let me know what should I do to pass the verify

**Comment by Yougoss** [comment]
Hi @slawekjaranowski 
Sorry, I forget to remove `-Drat.numUnapprovedLicenses=100` when I run `mvn clean verify` in my local.

I have fixed the issue, and it should pass the verify now

File: maven-resolver-provider/src/test/java/org/apache/maven/repository/internal/ArtifactDescriptorReaderDelegateTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package org.apache.maven.repository.internal;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.DependencyManagement;
+import org.apache.maven.model.Model;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.resolution.ArtifactDescriptorRequest;
+import org.eclipse.aether.resolution.ArtifactDescriptorResult;
+
```
Comment (slawekjaranowski) [inline maven-resolver-provider/src/test/java/org/apache/maven/repository/internal/ArtifactDescriptorReaderDelegateTest.java:9]: please put imports after license.

Comment (Yougoss) [reply maven-resolver-provider/src/test/java/org/apache/maven/repository/internal/ArtifactDescriptorReaderDelegateTest.java]: done

**Comment by Yougoss** [comment]
Hi @slawekjaranowski what should I do to move forward?

**Comment by michael-o** [comment]
We need to take a look at this next month and consider it for 3.9.x+.

**Comment by Yougoss** [comment]
Hi @michael-o , if we don't need it in 3.8.x, I will close this PR, since we have a same one for branch 3.9.x branch

**Comment by michael-o** [comment]
Let's close this out and focus and 3.9, then 4.0.

**Comment by jira-importer** [comment]
Resolve #8951

**Pull request closed without merging**